### PR TITLE
Functor.as/void

### PR DESCRIPTION
**Pull request by cquiroz**
Mechanical refactoring of places where `Functor.as` or `Functor.void` can be used instead of `map`

### CODE REVIEW

File: modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala [hunk 1]
```diff
@@ -82,7 +82,7 @@ class Engine[D, U](stateL: Engine.State[D]) {
           case r                 =>
             singleRunFailed(c, Result.Error(s"Unhandled result for single run action: $r"))
         }
-      ).map[EventResult.Outcome](_ => EventResult.Ok)
+      ).as[EventResult.Outcome](EventResult.Ok)
```
Comment (sraaphorst) [inline modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala]: What, out of curiosity, is the advantage of using `Functor.as` over `map`?

EDIT: Never mind. After scrolling through, I can see the obvious advantages. I didn't know about this before. Good to learn something new!

File: modules/seqexec/engine/src/test/scala/seqexec/engine/packageSpec.scala [hunk 2]
```diff
@@ -406,7 +406,7 @@ class packageSpec extends FlatSpec with NonImplicitAssertions {
 
     val c = ActionCoordsInSeq(stepId, ExecutionIndex(0), ActionIndex(0))
     val event = Event.modifyState[executionEngine.ConcreteTypes](
-      executionEngine.startSingle(ActionCoords(seqId, c)).map(_ => ())
+      executionEngine.startSingle(ActionCoords(seqId, c)).void
```
Comment (sraaphorst) [inline modules/seqexec/engine/src/test/scala/seqexec/engine/packageSpec.scala]: 👍 Much cleaner.

File: modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala [hunk 3]
```diff
@@ -277,7 +277,7 @@ class SeqexecEngine(httpClient: Client[IO], settings: Settings[IO], sm: SeqexecM
   def addSequencesToQueue(q: EventQueue, qid: QueueId, seqIds: List[Observation.Id])
   : IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
     Event.modifyState[executeEngine.ConcreteTypes](addSeqs(qid, seqIds)
-      .map[executeEngine.ConcreteTypes#EventData](_ => UpdateQueueAdd(qid, seqIds)))
+      .as[executeEngine.ConcreteTypes#EventData](UpdateQueueAdd(qid, seqIds)))
```
Comment (sraaphorst) [inline modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala]: Why this change? Do we want to always evaluate `UpdateQueueAdd`? Or will that not happen?

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/RemoteRequestsHandler.scala [hunk 4]
```diff
@@ -97,7 +97,7 @@ class RemoteRequestsHandler[M](modelRW: ModelRW[M, Option[ClientId]])
         Effect(
           SeqexecWebClient
             .resumeObs(id, step)
-            .map(_ => RunObsPause(id))
+            .as(RunObsPause(id))
```
Comment (sraaphorst) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/RemoteRequestsHandler.scala]: All these: same comment as above.

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/SequenceExecutionHandler.scala [hunk 5]
```diff
@@ -25,25 +25,31 @@ class SequenceExecutionHandler[M](
...
-          s.copy(metadata = s.metadata.copy(observer = Some(name)))
-        case s                        => s
-      })
+      val updateObserverE = Effect(
+        SeqexecWebClient.setObserver(sequenceId, name.value).as(NoAction))
+      val updatedSequences =
+        value.copy(sessionQueue = value.sessionQueue.collect {
+          case s if s.id === sequenceId =>
+            s.copy(metadata = s.metadata.copy(observer = Some(name)))
+          case s => s
```
Comment (sraaphorst) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/SequenceExecutionHandler.scala]: Not a big deal, but it looks like the spacing alignment was thrown off by your IDE.

File: modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala (hunk 3 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala]: `UpdateQueueAdd` is an output event, that tells the client that a sequence was added to a queue.
